File: CS/DXApplication1/Form1.cs
#region #mailmerge1
using DevExpress.Spreadsheet;
using DevExpress.XtraBars.Ribbon;
using System.Collections.Generic;

namespace DXApplication1 {
    public partial class Form1 : RibbonForm {

        nwindDataSet dataSet;
        nwindDataSetTableAdapters.CategoriesTableAdapter adapter;
        IWorkbook template;

        public Form1() {
            InitializeComponent();

            dataSet = new nwindDataSet();
            adapter = new nwindDataSetTableAdapters.CategoriesTableAdapter();
            adapter.Fill(dataSet.Categories);
        }

        private void Form1_Load(object sender, System.EventArgs e) {
            spreadsheetControl1.LoadDocument("Documents\\MailMergeTemplate.xlsx");
            template = spreadsheetControl1.Document;
            template.MailMergeDataSource = dataSet;
            template.MailMergeDataMember = "Categories";
        }
        #endregion #mailmerge1
        #region #mailmerge2
        private void barButtonItem1_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            IList<IWorkbook> resultWorkbooks = spreadsheetControl1.Document.GenerateMailMergeDocuments();
            int index = 0;
            foreach (IWorkbook workbook in resultWorkbooks) {
                string fileName = string.Format("Documents\\SavedDocument{0}" + ".xlsx", index++);
                workbook.SaveDocument(fileName, DocumentFormat.OpenXml);
            }
            System.Diagnostics.Process.Start("explorer.exe", "Documents\\");
        }
        #endregion #mailmerge2
    }
}

File: VB/DXApplication1/Form1.vb
Imports DevExpress.Spreadsheet
Imports DevExpress.XtraBars.Ribbon
Imports System.Collections.Generic

Namespace DXApplication1
    Partial Public Class Form1
        Inherits RibbonForm
#Region "#mailmerge1"
        Private dataSet As nwindDataSet
        Private adapter As nwindDataSetTableAdapters.CategoriesTableAdapter
        Private template As IWorkbook

        Public Sub New()
            InitializeComponent()

            dataSet = New nwindDataSet()
            adapter = New nwindDataSetTableAdapters.CategoriesTableAdapter()
            adapter.Fill(dataSet.Categories)
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            spreadsheetControl1.LoadDocument("Data\MailMergeTemplate.xlsx")
            template = spreadsheetControl1.Document
            template.MailMergeDataSource = dataSet
            template.MailMergeDataMember = "Categories"
        End Sub
#End Region ' #mailmerge1
        #Region "#mailmerge2"
        Private Sub barButtonItem1_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barButtonItem1.ItemClick
            Dim resultWorkbooks As IList(Of IWorkbook) = spreadsheetControl1.Document.GenerateMailMergeDocuments()
            Dim index As Integer = 0
            For Each workbook As IWorkbook In resultWorkbooks
                Dim fileName As String = String.Format("Data\SavedDocument{0}" & ".xlsx", index)
                index += 1
                workbook.SaveDocument(fileName, DocumentFormat.OpenXml)
            Next workbook
            System.Diagnostics.Process.Start("explorer.exe ", Application.StartupPath + "\Data")
        End Sub
        #End Region ' #mailmerge2
    End Class
End Namespace
